--- spectacle/Makefile	2018-11-20 14:37:35.848508428 -0500
+++ Makefile	2018-11-20 14:43:24.090652167 -0500
@@ -1,6 +1,6 @@
 CC=g++
-CFLAGS=-Wall -O3 -std=c++11 -I ./boost/include
-LDFLAGS=-std=c++11 ./boost/lib/libboost_iostreams.a ./zlib/install/lib/libz.a
+CFLAGS=-Wall -O3 -std=c++11 -I /usr/include 
+LDFLAGS=-std=c++11 -lboost_filesystem -lboost_system -lboost_iostreams -lz
 SRC_DIR=src
 LIB_DIR=lib
 BIN_DIR=bin
